pytorch_lightning/loops/utilities.py

```diff
@@ -62,6 +64,41 @@ def _extract_hiddens(training_step_output: STEP_OUTPUT, truncated_bptt_steps: in
     return hiddens
 
 
+def _parse_loop_limits(
+    min_steps: Optional[int],
+    max_steps: int,
+    min_epochs: Optional[int],
+    max_epochs: int,
+    max_time: Optional[Union[str, timedelta, Dict[str, int]]],
+    default_max_epochs: int = 1000,
+) -> Tuple[Optional[int], int]:
+    """This utility compute the number minimum and maximum number of epochs " based on minimum and maximum number
+    of steps and maximum time.
+
+    Args:
+        min_steps: Minimum number of steps
+        max_steps: Maximum number of steps
+        min_epochs:  Minimum number of epochs
+        max_epochs:  Maximum number of epochs
+        max_time: Maximum number of time for the training.
+        default_max_epochs: Default value for the number of max_epochs.
+
+    Returns:
+        A list of tuples (opt_idx, optimizer) of currently active optimizers.
+    """
+    if max_epochs is None:
+        if max_steps == -1 and max_time is None:
+            rank_zero_warn(
+                "`max_epochs` was not set. Setting it to 1000 epochs.",
+                UserWarning,
+            )
+            max_epochs = default_max_epochs
```

Comment:
```suggestion
) -> Tuple[Optional[int], int]:
    """This utility computes the number minimum and maximum number of epochs " based on the minimum and maximum number
    of steps and maximum time.

    Args:
        min_steps: Minimum number of steps
        max_steps: Maximum number of steps
        min_epochs: Minimum number of epochs
        max_epochs: Maximum number of epochs
        max_time: Maximum time for the training.

    Returns:
        A list of tuples (opt_idx, optimizer) of currently active optimizers.
    """
    if max_epochs is None:
        if max_steps == -1 and max_time is None:
            rank_zero_warn(
                "`max_epochs` was not set. Setting it to 1000 epochs.",
                UserWarning,
            )
            max_epochs = 1000
```
no need for the default argument.